### PR TITLE
Fix windows system memory lookup

### PR DESCRIPTION
**Pull request by dhiltgen**
This refines the gpu package error handling and fixes a bug with the system memory lookup on windows.

### CODE REVIEW

File: gpu/gpu_info_cpu.c [hunk 1]
```diff
@@ -6,11 +6,12 @@
 void cpu_check_ram(mem_info_t *resp) {
   resp->err = NULL;
   MEMORYSTATUSEX info;
+  info.dwLength = sizeof(info);
```
Comment (dhiltgen) [inline gpu/gpu_info_cpu.c:9]: fwiw, this was the *actual* fix.  The rest was just cleanups to get to a point where I could figure out this was the problem

File: gpu/gpu_test.go [hunk 2]
```diff
@@ -23,4 +23,20 @@ func TestBasicGetGPUInfo(t *testing.T) {
 	}
 }
 
+func TestCPUMemInfo(t *testing.T) {
+	info, err := getCPUMem()
+	assert.NoError(t, err)
+	switch runtime.GOOS {
+	case "darwin":
+		// TODO - remove this once MacOS returns some size for CPU
+		return
```
Comment (zeroZshadow) [inline gpu/gpu_test.go:32]: Instead of returning, should this be using `t.Skip` or similar?

File: gpu/gpu_info_cpu.c (hunk 1 above)
Comment (BruceMacD) [reply gpu/gpu_info_cpu.c:9]: Just to verify my understanding here. Setting dwLength here allows `info.ullTotalPhys` and `info.ullAvailPhys` to be calculated correctly by the call to `GlobalMemoryStatusEx` which takes the pointer to the info?

**Review by jmorganca — APPROVED** [review]

File: gpu/gpu_info_cpu.c (hunk 1 above)
Comment (dhiltgen) [reply gpu/gpu_info_cpu.c:9]: Correct.  I didn't look closely enough at the API docs and realize it needs to know the size of the input struct to operate properly.

**Pull request merged**